ui/app/src/App.tsx

```diff
@@ -61,8 +67,50 @@ function App() {
     return Math.random().toString(36).substring(7);
   };
 
-  // Get stream from CreateStream to pass it as props
+  const publicClient = usePublicClient();
+  const walletClient = useWalletClient()?.data;
+  const { address } = useAccount();
+
   const [stream, setStream] = useState<Stream | null>(null);
+  const [SM, setSM] = useState<StreamManager | null>(null);
+  const [createdStreams, setCreatedStreams] = useState<Stream[]>([]);
+
+  // Manage callback from onstreamcreate
+  const handleStreams = (newStream: Stream) => {
+      setCreatedStreams((prevStreams) => [...prevStreams, newStream]);
+  };
+
+  // Fetch the StreamManager and its streams
+  useEffect(() => {
+    const fetchStreamManager = async () => {
+      if (SM === null) {
+        try {
+          const newSM = await StreamManager.fromAddress(
+            config.streamManagerAddress as `0x${string}`,
+            publicClient,
+            walletClient as WalletClient
+          );
+          setSM(newSM);
+        } catch (error) {
+          console.error("Stream Manager Error:", error);
+        }
+      }
+    };
+    fetchStreamManager();
+
+    // Fetch create streams when SM has been fetched
+    const interval: NodeJS.Timeout = setInterval(() => {
+      if (SM != null && createdStreams.length === 0) {
+        SM.onStreamCreated(handleStreams, address);
```

Comment:
this installs a watcher, so you shouldn't have to do polling because it should do it already (hopefully via websockets)

Reply:
in which case, we can probably chain off the promise above and make the `.then` handler a bit more complex to install this when it sets the `SM` state variable